validator/db/kv/attestation_history.go

```diff
@@ -81,20 +110,98 @@ func (store *Store) SaveAttestationHistoryForPubKeys(
 				return err
 			}
 		}
+		for pubKey, min := range minByPubKeys {
+			if err := bucket.Put(attHist.GetMinSourceKey(pubKey), bytesutil.Uint64ToBytesBigEndian(min.Source)); err != nil {
+				return err
+			}
+			if err := bucket.Put(attHist.GetMinTargetKey(pubKey), bytesutil.Uint64ToBytesBigEndian(min.Target)); err != nil {
+				return err
+			}
+		}
 		return nil
 	})
 	return err
 }
 
 // SaveAttestationHistoryForPubKey saves the attestation history for the requested validator public key.
 func (store *Store) SaveAttestationHistoryForPubKey(
-	ctx context.Context, pubKey [48]byte, history attHist.History,
+	ctx context.Context,
+	pubKey [48]byte,
+	history attHist.History,
 ) error {
 	ctx, span := trace.StartSpan(ctx, "Validator.SaveAttestationHistoryForPubKey")
 	defer span.End()
 	err := store.update(func(tx *bolt.Tx) error {
 		bucket := tx.Bucket(newHistoricAttestationsBucket)
-		return bucket.Put(pubKey[:], history)
+		if err := bucket.Put(pubKey[:], history); err != nil {
+			return err
+		}
+
+		return nil
 	})
 	return err
 }
+
+// SaveMinAttestation saves min attestation values if they are lower from the ones that are currently set in db.
+func (store *Store) SaveMinAttestation(ctx context.Context, pubKey [48]byte, minAtt attHist.MinAttestation) error {
+	ctx, span := trace.StartSpan(ctx, "Validator.SaveMinAttestation")
+	defer span.End()
+
+	return store.update(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(newHistoricAttestationsBucket)
+		minSourceStorageKey := attHist.GetMinSourceKey(pubKey)
+		enc := bucket.Get(minSourceStorageKey)
+		if len(enc) == 0 {
+			minSource := bytesutil.BytesToUint64BigEndian(enc)
+			if minAtt.Source < minSource {
+				if err := bucket.Put(minSourceStorageKey, bytesutil.Uint64ToBytesBigEndian(minAtt.Source)); err != nil {
+					return err
+				}
+			}
+		} else {
+			if err := bucket.Put(minSourceStorageKey, bytesutil.Uint64ToBytesBigEndian(minAtt.Source)); err != nil {
+				return err
+			}
+		}
+		minTargetStorageKey := attHist.GetMinTargetKey(pubKey)
+		enc = bucket.Get(minTargetStorageKey)
+		if len(enc) == 0 {
+			minTarget := bytesutil.BytesToUint64BigEndian(enc)
+			if minAtt.Target < minTarget {
+				if err := bucket.Put(minTargetStorageKey, bytesutil.Uint64ToBytesBigEndian(minAtt.Target)); err != nil {
+					return err
+				}
+			}
+		} else {
+			if err := bucket.Put(minTargetStorageKey, bytesutil.Uint64ToBytesBigEndian(minAtt.Source)); err != nil {
+				return err
+			}
+		}
+		return nil
```

Comment:
I'm curious about this logic. Do we care about the earliest attestation signed as a whole, or both earliest source signed and earliest target signed? If so, we need to perhaps clarify that in comments